engine/tests/phpunit/integration/Elgg/Database/DelayedEmailQueueTableIntegrationTest.php

```diff
@@ -136,6 +131,31 @@ public function testDeleteRecipientRows() {
 		$this->assertEmpty($this->table->getRecipientRows($recipient->guid, 'weekly', $dt->getTimestamp()));
 	}
 	
+	public function testDeleteAllRecipientRows() {
+		$notification = $this->getTestNotification();
+		$recipient = $notification->getRecipient();
+		
+		// insert
+		for ($i = 0; $i < 5; $i++) {
+			$this->assertTrue($this->table->queueEmail($recipient->guid, 'daily', $notification));
+		}
+		
+		// different interval
+		for ($i = 0; $i < 5; $i++) {
+			$this->assertTrue($this->table->queueEmail($recipient->guid, 'weekly', $notification));
+		}
+		
+		$dt = $this->table->getCurrentTime('+10 seconds');
+		
+		// delete
+		$recipient_guid = $recipient->guid;
+		$recipient->delete();
```

Comment:
This is a far reliance on the integration of entity->delete will trigger the cleanup of this table. It would be better the just stick to testing this table in this test.
If you wish to test that entity->delete will clean the table, write that in the integration test of an entity

Reply:
changed it and added a separate test for cleanup during entity->delete